File: Leetcode/python/sol3005.py
#  https://leetcode.com/problems/count-elements-with-maximum-frequency/description/?envType=problem-list-v2&envId=hash-table
class Solution:
    def count(self, arr :list[int]) -> dict[int, int]:
        counter :dict[int ,int] = {}
        for i in arr:
            counter[i] = counter.get(i, 0) + 1
        return counter

    def maxFrequencyElements(self, nums: List[int]) -> int:
        counter :list[int] = list(self.count(nums).values())
        max_freq :int = max(counter)
        return sum(i for i in counter if i ==  max_freq)

File: Leetcode/cpp/sol3005.cpp
//https://leetcode.com/problems/count-elements-with-maximum-frequency/?envType=daily-question&envId=2025-09-22

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> count;
        for(auto i:nums){
            count[i]++;
        }
        int max_freq = 0;
        for (auto i:count){
            max_freq = max(max_freq, i.second);
        }
        int result = 0;
        for (auto i:count){
            if (i.second == max_freq){
                result+= i.second;
            }
        }

        return result;
    }
};